nats.go

```diff
@@ -3089,6 +3181,91 @@ func (nc *Conn) IsConnected() bool {
 	return nc.isConnected()
 }
 
+// drainConnection will run in a separate Go routine and will
+// flush all publishes and drain all active subscriptions.
+func (nc *Conn) drainConnection() {
+	// Snapshot subs list.
+	nc.mu.Lock()
+	subs := make([]*Subscription, 0, len(nc.subs))
+	for _, s := range nc.subs {
+		subs = append(subs, s)
```

Comment:
Since we can't Drain a non async subscription, should we select only async subscriptions here? But if that's the case, then checking on `NumSubscriptions() == 0` is kind of wrong.

Reply:
No because this makes sure the messages are in the channel for sync and they can do what they want with those messages.